pkg/management/client.go

```diff
@@ -132,17 +132,22 @@ func NewEventRecorder() (record.EventRecorder, error) {
 // WaitKubernetesAPIServer will wait for the kubernetes API server to by ready.
 // Returns any error if it can't be reached.
 func WaitKubernetesAPIServer(ctx context.Context, clusterObjectKey client.ObjectKey) error {
-	logger := log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithName("wait-kubernetes-api-server")
 
 	cli, err := NewControllerRuntimeClient()
 	if err != nil {
 		logger.Error(err, "error while creating a standalone Kubernetes client")
 		return err
 	}
 
-	if err := retry.OnError(readinessCheckRetry, resources.RetryAlways, func() (err error) {
-		return cli.Get(ctx, clusterObjectKey, &apiv1.Cluster{})
-	}); err != nil {
+	err = retry.OnError(readinessCheckRetry, resources.RetryAlways, func() error {
+		if err := cli.Get(ctx, clusterObjectKey, &apiv1.Cluster{}); err != nil {
+			logger.Warning("Encountered an error while executing get cluster. Will wait and retry", "error", err.Error())
+			return err
```

Comment:
I modified the log message because it could end up being misleading, we are exiting in case of any error, which could be a lack of permission or cluster not existing anymore, for example (even if unlikely)

Reply:
perhaps WaitForGetCluster would be more meaningful?